### PR TITLE
Added screenshots on test failure

### PR DESCRIPTION
**Pull request by jossjacobo**
Added a test watcher that will take a screenshot if the test fails and
the config property is set to true. I also added config values to readme
to make it a bit more obvious what you can configure.

Added screenshotOnFail config param to readme

### CODE REVIEW

File: src/main/java/com/joss/conductor/mobile/Locomotive.java [hunk 1]
```diff
@@ -163,9 +168,45 @@ private IOSDeviceUtil iosDeviceUtil() {
...
+        @Override
+        protected void finished(Description description) {
+            super.finished(description);
+            if (configuration.screenshotsOnFail()) {
+                if (failure) {
+                    ScreenShotUtil.take(Locomotive.this,
+                            description.getDisplayName(),
+                            e.getMessage());
+                }
+                Locomotive.this.driver.quit();
```
Comment (imbeyondboredom) [inline src/main/java/com/joss/conductor/mobile/Locomotive.java]: Do you want this to quit if it didn't fail?

Comment (jossjacobo) [inline src/main/java/com/joss/conductor/mobile/Locomotive.java]: Yes, if the featured is enabled `finished` is in charge of quitting the driver, regardless of whether the test failed or not.

File: src/main/java/com/joss/conductor/mobile/util/ScreenShotUtil.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    public static void take(Locomotive locomotive, String path, String testName) {
+        try {
+            FileUtils.copyFile(locomotive.driver.getScreenshotAs(OutputType.FILE),
+                    new File(WORKING_DIR
+                            + File.separator
+                            + DIR
+                            + File.separator
+                            + path.replace(File.separator, "-")
+                            + File.separator
+                            + testName.replace(File.separator, "-") + ".png"));
```
Comment (imbeyondboredom) [inline src/main/java/com/joss/conductor/mobile/util/ScreenShotUtil.java]: If a File.separator can be in the test name then can other symbols that aren't allowed as a file name be included?

**Comment by jossjacobo** [comment]
@imbeyondboredom I added a few checks for the filename creation, it turns out it IS a bug :) wanna give it quick look?

File: src/main/java/com/joss/conductor/mobile/util/ScreenShotUtil.java [hunk 3]
```diff
@@ -14,33 +15,55 @@
...
+                + DIR
+                + File.separator
+                + limitLengthTo100Chars(removeInvalidFilenameChars(path))
+                + File.separator
+                + limitLengthTo100Chars(removeInvalidFilenameChars(testName))
+                + PNG_EXT;
+    }
+
+    private static String removeInvalidFilenameChars(String name) {
+        return name.replace(File.separator, "-")
```
Comment (imbeyondboredom) [inline src/main/java/com/joss/conductor/mobile/util/ScreenShotUtil.java]: The only thing to consider here is that if you have "thisTest*" and "thisTest?" it'll end up being the same files. If that's common enough you might want to add a timestamp to the end of the filename

File: src/main/java/com/joss/conductor/mobile/util/ScreenShotUtil.java [hunk 4]
```diff
@@ -14,33 +15,55 @@
...
+                .replace(":", "-")
+                .replace("?", "-")
+                .replace("*", "-")
+                .replace("|", "-")
+                .replace(":", "-");
```
Comment (imbeyondboredom) [inline src/main/java/com/joss/conductor/mobile/util/ScreenShotUtil.java]: Is this supposed to be a semicolon? You've already replaced a colon a few lines above so this is unnecessary

**Pull request merged**